### ISSUE TITLE
Add neural network output module 

### ISSUE THREAD

**Issue by agibsonccc** [eclipse/deeplearning4j#1800]
To avoid polluting the core model api with too much domain specific logic we should add a module which allows us to add features relating to neural net apps (eg: output ).

The example that came up here is label names on the predict method. This will be a great module for community contributions as well.

**Comment by AlexDBlack** [comment]
Another thing that could go in here: extracting probabilities from sequence classification. The index of the classification can vary (depending on the length of the time series) but we can obtain that index for each sequence from the mask array.

An additional idea: decomposing loss functions. For example, MSE can be broken down into components/contributions from each network output. This might be useful for model interpretability.

**Comment by raver119** [comment]
We have that, it's OutputAdapter/ModelAdapter interfaces.
